docs/detections/session-view.asciidoc

```diff
@@ -84,4 +90,40 @@ Session View includes an additional badge not pictured above:
 * The *Exec user change* badge highlights exec user changes, such as when a user escalates to root:
 +
 [role="screenshot"]
-image::images/session-view-exec-user-change-badge.png[The Exec user change button,width=80%,height=80%]
+image::images/session-view-exec-user-change-badge.png[The Exec user change badge,width=80%,height=80%]
+
+* The *Output* badge appears next to commands that generated terminal output. Click it to open the terminal output view.
++
+[role="screenshot"]
+image::images/session-view-output-badge.png[The Output badge,width=80%,height=80%]
+
+[[session-view-output]]
+[discrete]
+=== Terminal output view UI
+
+beta::[]
+
+Terminal output view provide a more complete understanding of commands executed by users or adversaries, and assists with auditing and compliance. It allows you to review a session's terminal output in the context of the original commands. It includes both terminal output and shell commands (if the shell was in echo mode).
+
+To enable terminal output data capture:
+
+. Go to *Manage* -> *Policies*, then select one or more of your {elastic-defend} integration policies to edit.
+. On the *Policy settings* tab, scroll down to the Linux event collection section near the bottom of the page
+and select the *Include session data* and *Capture terminal output* options.
+
+You can configure several additional settings by clicking *Advanced settings* at the bottom of the page:
+
+* `linux.advanced.tty_io.max_kilobytes_per_process`: The maximum number of kilobytes of output to record from a single process. Default: 512 KB. Process output exceeding this value will not be recorded.
+* `linux.advanced.tty_io.max_kilobytes_per_event`: The maximum number of kilobytes of output to send to {es} as a single event. Default: 512 KB. Additional data is captured as a new event.
+* `linux.advanced.tty_io.max_event_interval_seconds`: The maximum interval (in seconds) during which output is batched. Default: 30 seconds. Output will be sent to {es} at this interval (unless it first exceeds the `max_kilobytes_per_event` value, in which case it might be sent sooner).
+
+[role="screenshot"]
+image::images/session-view-output-viewer.png[Terminal output view]
+
+1. Search bar. Use to find and highlight search terms within the current session.
+The left and right arrows allow you to navigate through search results.
```

Comment:
Detail on why search may behave unexpectedly would help as well as emphasis that this is only search over terminal output held in the browser (ie it is not an Elasticsearch query - @mitodrummer pls correct me if I'm wrong).  E.g.:

"Use to find and highlight search terms within the current session's terminal output held by the browser.   Note that because terminal output may include terminal codes for colorizing the text and moving the cursor, some searches may not find a match.  For example, if one is searching output for a command like "sudo" (echoed command input) and the user mis-entered "sudo" as "sdo", and moved the cursor back to insert the missing "u", the terminal output search for "sudo" will not match.  This challenge can also be present in visual editors like vi and emacs where cursor movement for editing is commonplace"

Reply:
You are correct @m-sample, though, for the browser side "plain text" search, I strip out all ansi chars on a a line before matching the user's search against it. This is required so that I get an accurate string index position for each match, so that when I seek to this "line" in the tty player, it's able to highlight the correct match (if there are multiple on a single line). That being said, you are correct in that if a user mistypes a command and then hits backspace to correct it, the resulting output text will not be the "corrected" version.

e.g
User types,    sdo
They hit backspace twice, then type udo to fix their typo.
The resulting output will look something simlilar to:
sdo\b\budo

The \b is an escape code for backspace, so you can see why searching for the corrected command will not match.

That being said, most of the formating ansi codes for color and text weight would be stripped out and should not interfere with most plain text searches. This is in contrast to using KQL or EQL to match on values in process.io.text as it will be the un altered version of the value that will potentially contain many escape sequences between words in the output text. Here is a link to a resource which describes these codes. https://gist.github.com/fnky/458719343aabd01cfb17a3a4f7296797